### DISCUSSION TITLE
HELP!!! Where LabelQuerier is imlemented?

### DISCUSSION THREAD

**Discussion by Tusenka** [prometheus/prometheus#15116] [Q&A] [answered]
Where is interface LabelQuerier is implemented
https://github.com/prometheus/prometheus/blob/5a54ae67edc4b140f14df25b3f08e438b5f846a6/storage/interface.go#L161

**Comment by bwplotka** [comment, marked as answer]
Advanced IDEs will give you this info quickly:
![image](https://github.com/user-attachments/assets/3bb2c9b4-514a-452b-b6c6-a34d1449190d)

There are 34 implementations, which one are you interested in?


![image](https://github.com/user-attachments/assets/56e363f7-878f-43eb-933c-cdb0c742d574)
![image](https://github.com/user-attachments/assets/2b09e4a5-1398-4914-9f73-0997696bbfaf)


Without editor, you can also:
* search for `LabelValues` method implementations
* check from the type that requires this interface, where it's initialized and how. This will tell you (eventually, perhaps in `main.go`) what implementation is passed to that type.

Hope that helps!
